vllm/engine/async_llm_engine.py

```diff
@@ -479,6 +491,37 @@ async def check_health_async(self) -> None:
             self.tokenizer.check_health()
         self.model_executor.check_health()
 
+    async def _build_guided_decoding_logits_processor_async(
+            self, sampling_params: SamplingParams,
+            lora_request: Optional[LoRARequest]) -> SamplingParams:
+        """Constructs logits processors based on the guided_decoding,
+        logits_bias, and allowed_token_ids fields in sampling_params. Deletes
+        those fields and adds the constructed logits processors to the
+        logits_processors field. Returns the modified sampling params."""
+        if (guided_decoding := sampling_params.guided_decoding) is None:
+            return sampling_params
+
+        logger.debug(
+            "Building guided decoding logits processor in "
+            "AsyncLLMEngine. Params: %s", guided_decoding)
+
+        tokenizer = self.get_tokenizer(lora_request=lora_request)
+        guided_decoding.backend = guided_decoding.backend or \
+            self.decoding_config.guided_decoding_backend
+
+        processor = await get_guided_decoding_logits_processor(
+            guided_params=guided_decoding, tokenizer=tokenizer)
+
+        if processor:
+            if sampling_params.logits_processors is None:
+                sampling_params.logits_processors = []
+            sampling_params.logits_processors.append(processor)
+
+        # Unset guided decoding params after constructing the lp from them
+        sampling_params.guided_decoding = None
+
+        return sampling_params
```

Comment:
@joerunde could we make this a utility method since it's duplicated in `client.py`? We could add an additional `get_tokenizer` callable arg.

Reply:
can do!